uaclient/contract.py

```diff
@@ -383,7 +383,7 @@ def process_entitlement_delta(
                 )
             )
         try:
-            ent_cls = ENTITLEMENT_CLASS_BY_NAME[name]
+            ent_cls = entitlement_factory(name)
         except KeyError:
```

Comment:
@renanrodrigo Looks like this will need to change so that instead of catching a `KeyError` it checks if `ent_cls` is None

Reply:
already done, hadn't tackled it yet in this WIP version (: